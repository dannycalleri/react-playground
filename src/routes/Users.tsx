import React from "react";
import { Link } from "react-router-dom";
import { UsersList } from "../components/UsersList";
import { users } from "../data/users";

export function Users() {
  return (
    <>
      <h1>Users</h1>
      <Link to={`/users/new`}>New User</Link>
      <UsersList users={users} />
    </>
  );
}
