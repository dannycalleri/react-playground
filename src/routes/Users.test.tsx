import { render } from "@testing-library/react";
import { Users } from "./Users";

jest.mock("react-router-dom", () => {
  return {
    Link: (props: any) => {
      const { to, children } = props;
      return (
        <>
          <a href={to}>{children}</a>
        </>
      );
    },
  };
});

describe("Users route", () => {
  it("should render correctly", () => {
    const { asFragment } = render(
      <Users
        state={{ data: [{ id: 1, name: "Sandro" }], dispatch: jest.fn() }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
