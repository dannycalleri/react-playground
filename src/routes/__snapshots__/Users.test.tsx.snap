// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users route should render correctly 1`] = `
<DocumentFragment>
  <h1>
    Users
  </h1>
  <a
    href="/users/new"
  >
    New User
  </a>
  <ul>
    <li>
      <a
        href="/users/1"
      >
        Sandro
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
